addons/addon-base-raas/packages/base-raas-services/lib/budgets/budgets-service.js

```diff
@@ -0,0 +1,265 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const _ = require('lodash');
+const moment = require('moment');
+const Service = require('@aws-ee/base-services-container/lib/service');
+const { runAndCatch } = require('@aws-ee/base-services/lib/helpers/utils');
+const { isAdmin, isActive } = require('@aws-ee/base-services/lib/authorization/authorization-utils');
+const budgetBaseTemplates = require('./budget-base-templates');
+const createSchema = require('../schema/create-budget');
+
+class BudgetsService extends Service {
+  constructor() {
+    super();
+    this.dependency(['aws', 'awsAccountsService', 'jsonSchemaValidationService']);
+    this.budgetBaseTemplates = budgetBaseTemplates;
+    this.budgetName = 'service-workbench-system-generated-budget-do-not-update';
+  }
+
+  async init() {
+    await super.init();
+    // Lazy initialization for services
+    this.aws = null;
+    this.awsAccountsService = null;
+    this.jsonSchemaValidationService = null;
+  }
+
+  async get(requestContext, id) {
+    await this._initServices();
```

Comment:
I believe changes for removing 'this' usage is still underway.

Reply:
Yes! Working on it now :)

Reply:
It's fixed!